src/useFetch.ts

```diff
@@ -175,10 +175,14 @@ function useFetch<TData = any>(...args: UseFetchArgs): UseFetch<TData> {
         controller.current = undefined
       }
 
-      if (newRes && !newRes.ok && !error.current) error.current = makeError(newRes.status, newRes.statusText)
+      if (newRes && (newRes.ok === false || newRes.status >= 400) && !error.current) {
+        error.current = makeError(newRes.status, newRes.statusText);
+      }
       if (!suspense) setLoading(false)
       if (attempt.current === retries) attempt.current = 0
 
+      if (error.current) throw error.current;
```

Comment:
I don't think we want to throw here. We want to return the `error`. i.e.
```js
const { error } = useFetch('https://httpbin.org/status/400')
console.log('error', error)
```
We want to be able to use this error^ not kill the page. Unless I'm misunderstanding what you're thinking here.

Reply:
When suspense mode, we need throw Error to trigger ErrorBoundary, or should modify other?

Reply:
Suspense throw is called at the bottom of `useFetch`. [See here](https://github.com/ava/use-http/blob/master/src/useFetch.ts#L258). Do the tests work without this?